drivers/gpio/gpio_sifive.c

```diff
@@ -117,25 +117,34 @@ static int gpio_sifive_config(struct device *dev,
 	}
 
 	/* Configure gpio direction */
-	if (flags & GPIO_DIR_OUT) {
+	if (flags & GPIO_OUTPUT) {
 		gpio->in_en &= ~BIT(pin);
 		gpio->out_en |= BIT(pin);
```

Comment:
It seems like on this platform input and output can be enabled / disabled independently. If this is the case we should write the code as follows
```
	if ((flags & GPIO_OUTPUT) != 0) {
￼		gpio->out_en |= BIT(pin);
	} else {
￼		gpio->out_en &= ~BIT(pin);
	}
```
or alternatively use `WRITE_BIT` macro
```
	WRITE_BIT(gpio->out_en, pin, flags & GPIO_OUTPUT);
```
We should do the same for input flag.